client/reader/share/index.jsx

```diff
@@ -43,27 +46,28 @@ var actionMap = {
 		window.open( fackbookUrl, 'facebook', 'width=626,height=436,resizeable,scrollbars' );
 	},
 	pressThis: function( post ) {
-		var primarySite = sitesList.getPrimary(), wordpressUrl;
-
+		const primarySite = sitesList.getPrimary();
 		if ( ! primarySite ) {
 			return;
 		}
 
-		if ( primarySite.wpcom_url && ! primarySite.jetpack ) {
-			wordpressUrl = url.parse( 'https://' + primarySite.wpcom_url + '/wp-admin/press-this.php' );
-		} else {
-			wordpressUrl = url.parse( url.resolve( primarySite.URL, 'wp-admin/press-this.php' ) );
+		const args = {};
+
+		if ( post.content_embeds && post.content_embeds.length ) {
+			args.embed = post.content_embeds[0].src;
```

Comment:
If there's a worry about invalid URLs and it's possible for multiple embeds to exist, should we perform the `isValidUrl` logic here so we can check all of the available options? i.e.

```js
args.embed = find( post.content_embeds, { src: isValidUrl( src ) } );
```